### PR TITLE
update labs blog to use the blog post image and alt for social cards

### LINKED ISSUE #738: [FEAT] Add blog metadata handling for per-post custom social cards
#733 implements a base framework for social card metadata in the form of the `SocialCard` component.

At this time, all pages on each site point to the same stock social card image, which is the logo (either Quansight or Labs) on a plain white field.

This issue requests machinery to be built atop the work of #733 to allow individual blog posts to define the following:

- Twitter social card image
- Twitter social card image alt-text
- Twitter social card *type* (either `summary`, with a small 1:1 image; or `summary_large_image`, with a larger, 1.91:1 image)
- OpenGraph social card image (1.91:1 aspect ratio; primarily for LinkedIn but usable by other consumers)
- OpenGraph social card image alt text

Custom social card metadata for other site pages will be tracked in a separate issue, because that machinery will differ significantly from the blog machinery, and will require augmentation of the Storyblok schemas.

This issue is subsequent to #733, which addresses #706.

This issue represents a portion of the original #431. 

---

_Original discussion downstream of a post by @gabalafou in https://github.com/Quansight/Quansight-website/pull/733#discussion_r1210355997_

**gabalafou**:

Maybe this is for a follow-up PR but I wonder for blog posts if we should do something like the following:

```
      <SocialCard
        title={post.meta.title}
        description={post.meta.description}
        variant={DomainVariant.Quansight}
        twitterCardImageUrl={`${process.env.NEXT_PUBLIC_VERCEL_URL}${post.meta.featuredImage.src}`}
        openGraphCardImageUrl={`${process.env.NEXT_PUBLIC_VERCEL_URL}${post.meta.featuredImage.src}`}
        cardImageAlt={`post.meta.featuredImage.alt}
      />
```

I'm guessing we'll have to do more work to make sure that the images are the right dimensions for Twitter and Open Graph, meaning I guess we might need to add new metadata fields for blog posts such as `post.meta.twitterImage`.

**bskinn**:

> I'm guessing we'll have to do more work to make sure that the images are the right dimensions for Twitter and Open Graph

Exactly this, Twitter is 1:1, OpenGraph is 1.91:1. I doubt either exactly matches the ratio for the featuredImage; and, for some posts, we will want to be able to specify images of the correct ratio.

**gabalafou**:

I think both Storyblok and Vercel have image resizing/cropping features... we might be able to use them here... but will take some looking into

**bskinn**:

Storyblok definitely does, though I don't think you can have one asset with multiple differently-cropped 'views' -- I think you get one cropping per asset.

So, we're looking at uploading multiple assets anyways.

**bskinn**:

Following up here; we're implementing so that we have the option for either the 1:1 `summary` Twitter card or the 1.9:1 `summary_large_image` card.

Thus, we'll have to also add metadata to indicate which type of Twitter card is desired.

### PR DESCRIPTION
**Pull request by Surafel-Getachew**
Closes #738.

### CODE REVIEW

**Comment by bskinn** [comment]
Looks like these need the explicit leading `https://`, @Surafel-Getachew.

**Comment by Surafel-Getachew** [comment]
[preview](https://socialsharepreview.com/?url=https://labs-japgvncte-quansight.vercel.app/blog/sangho-internship-blogpost)

**Comment by bskinn** [comment]
Twitter is working:

image

LinkedIn is working:

image


@gabalafou, I think this may be GTG. WDYT?

**Review by gabalafou — APPROVED** [review]
Sweet! Thanks @Surafel-Getachew 

🌻

File: README.md [hunk 1]
```diff
@@ -547,12 +547,14 @@ signs. The meta section contains post-related information like:
...
   **Important note:** categories are case-sensitive.
-- `featuredImage` (required) - Object with two required properties: `src` and `alt`. The `src` property
-  is a path to the featured image which is displayed on the posts list on the`/blog`
-  page. The `alt` property is alternative text for the image. The image should
-  be added to the `apps/labs/public/posts/<post-name>` directory, for example,
-  `apps/labs/public/posts/hello-world-post`. There is no need to provide a full image path,
-  so the path name should start with `/posts/`.
+- `featuredImage` (required) - Object with two required properties: `src` and `alt`.
+  - The `src` property
+    is a path to the featured image which is displayed both (a) on the posts gallery on the`/blog`
```
Comment (bskinn) [inline README.md:552]:
```suggestion
    is a path to the featured image which is displayed both (a) in the posts gallery on the`/blog`
```

**Pull request merged**